src/napari_spatialdata/_interactive.py

```diff
@@ -77,7 +80,36 @@ def _onClick(self, text: str) -> None:
         elif self.elements_widget._elements[text] == "shapes":
             self._add_shapes(text)
 
+    def _update_visible_in_cs(self, event: Event) -> None:
+        """Toggle active in cs metadata when changing visibility of layer."""
+        layer = event.source
+        layer_active = layer.metadata["active_in_cs"]
+        selected_coordinate_system = self.coordinate_system_widget._system
+
+        elements = self.elements_widget._elements
+        if layer.name in elements:
+            if selected_coordinate_system not in layer_active:
+                layer_active.add(selected_coordinate_system)
+            else:
+                layer_active.remove(selected_coordinate_system)
+
+    def _update_layers_visibility(self) -> None:
+        """Toggle layer visibility dependent on presence in currently selected coordinate system."""
+        elements = self.elements_widget._elements
+        coordinate_space = self.coordinate_system_widget._system
+
+        # No layer selected on first time coordinate system selection
+        if self._viewer.layers:
+            for layer in self._viewer.layers:
+                if layer.name not in elements:
+                    layer.visible = False
+                elif layer.metadata["active_in_cs"]:
+                    layer.visible = True
+                    # Prevent _update_visible_in_cs of removing coordinate system if previously in coordinate system.
```

Comment:
This here ensures that the coordinate system in which a layer is visible is not removed when changing back to a previously selected coordinate system in which the layer is present and was visible. Otherwise in `_update_visible_in_cs`

```
else:
      layer_active.remove(selected_coordinate_system)
```
This would ensure that the coordinate system would stay removed from the metadata, which should not happen.